labels/index.ts

```diff
@@ -0,0 +1,105 @@
+import './style.css';
+import 'maplibre-gl/dist/maplibre-gl.css';
+import {Deck} from '@deck.gl/core';
+import maplibregl from 'maplibre-gl';
+import {VectorTileLayer, vectorQuerySource} from '@deck.gl/carto';
+import {CollisionFilterExtension} from '@deck.gl/extensions';
+import {scaleLinear} from 'd3-scale';
+
+const apiBaseUrl = import.meta.env.VITE_API_BASE_URL;
+const accessToken = import.meta.env.VITE_API_ACCESS_TOKEN;
+const connectionName = 'carto_dw';
+const cartoConfig = {apiBaseUrl, accessToken, connectionName};
+
+const INITIAL_VIEW_STATE = {
+  latitude: 39.1,
+  longitude: -94.57,
+  zoom: 3.8,
+  maxZoom: 16,
+  pitch: 0,
+  bearing: 0
+};
+
+const colorScale = scaleLinear()
+  .domain([3, 4, 5, 6, 7])
+  .range([
+    [29, 145, 192],
+    [65, 182, 196],
+    [127, 205, 187],
+    [199, 233, 180],
+    [237, 248, 177]
+  ]);
+
+  const fontSize = 24;
+  const scale = 16;
+  const sizeMaxPixels = (scale / 3) * fontSize;
+  const sizeMinPixels = Math.min(scale / 1000, 0.5) * fontSize;
+
+const MAP_STYLE = 'https://basemaps.cartocdn.com/gl/dark-matter-nolabels-gl-style/style.json';
+
+const deck = new Deck({
+  canvas: 'deck-canvas',
+  initialViewState: INITIAL_VIEW_STATE,
+  controller: true
+});
+
+// Add basemap
+const map = new maplibregl.Map({
+  container: 'map',
+  style: MAP_STYLE,
+  interactive: false
+});
+
+deck.setProps({
+  onViewStateChange: ({viewState}) => {
+    const {longitude, latitude, ...rest} = viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+  }
+});
+
+async function render() {
+  const dataSource = vectorQuerySource({
+    ...cartoConfig,
+    sqlQuery: `SELECT geom, population, name 
+      FROM cartobq.public_account.cities_1000 
+        WHERE population > 0
+        ORDER BY population DESC`
```

Comment:
This requires more explanation in a comment. I think ORDER BY is required because Maps API drops points in the same location and the ORDER BY is used to control that dropping. Is that right?

Reply:
Yes, I believe that is correct

Reply:
`WHERE population > 0` is to avoid collisions with locations that have a population equal 0
`ORDER BY`is to prioritize the biggest cities before the smallest

Reply:
`ORDER BYis to prioritize the biggest cities before the smallest`

This is not correct, we use getCollisionPriority for that purpose, you need the ORDER BY to guarantee the maps api is not snapping/dropping the wrong points.